lib/transformer.dart

```diff
@@ -84,14 +85,14 @@ class InitializeTransformer extends Transformer {
     if (primaryId == null) primaryId = transform.primaryInput.id;
     var newEntryPointId = new AssetId(primaryId.package,
         '${path.url.withoutExtension(primaryId.path)}.initialize.dart');
-    return transform.hasInput(newEntryPointId).then((exists) {
+    return transform.hasInput(newEntryPointId).then((exists) async {
```

Comment:
async/await shouldn't be necessary here either, my guess is this is necessary to get around issues with `Future` flattening? Were you seeing errors about `Future<Future<X>>`?